extensions/wikia/Chat2/ChatAjax.class.php

```diff
@@ -153,48 +171,52 @@ static public function getPrivateRoomID() {
 	 * @throws BadRequestException
 	 */
 	static public function blockOrBanChat() {
-		ChatHelper::info( __METHOD__ . ': Method called' );
 		global $wgRequest, $wgUser;
+
 		wfProfileIn( __METHOD__ );
+		Chat::info( __METHOD__ . ': Method called' );
+
+		if ( !self::authenticateServerOrUser() ) {
+			wfProfileOut( __METHOD__ );
 
-		// MAIN-6290  server.js needs to pass edit token for this to work
-		// $wgRequest->isValidWriteRequest( $wgUser );
+			return [ 'error' => self::ERROR_NOT_AUTHENTICATED ];
+		}
 
 		$kickingUser = $wgUser;
 
-		$retVal = array();
+		$res = [ ];
 		$userToBan = $wgRequest->getVal( 'userToBan' );
 		$userToBanId = $wgRequest->getVal( 'userToBanId', 0 );
 
 		if ( !empty( $userToBanId ) ) {
 			$userToBan = User::newFromId( $userToBanId );
-			if ( !empty( $userToBanId ) ) {
-				$userToBan = $userToBan->getName();
-			}
+			$userToBan = $userToBan->getName();
```

Comment:
Is there any reason we can't just do $userToBan = User::newFromId( $userToBanId )->getName();? I thought they fixed this in current php versions.